api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -174,7 +181,7 @@ public void setDistinctWorkspaceLimit(int limit) {
             .limit(distinctWorkspacelimit)
             .collect(Collectors.toList());
 
-    final ImmutableMap<Long, FirecloudWorkspaceResponse> idToLiveWorkspace =
+    final Map<Long, DbWorkspace> idToDbWorkspace =
```

Comment:
I'm not opposed to using immutables. This was in the interest of using the least strict type which in this case is Map. IMO, immutability is an implementation detail.

Reply:
In the case of the Immutable variants, the general preferred style is actually to use the more specific `Immutable` type where possible. The reason is that it has a semantically meaningful distinction in its interface - it's not just an implementation detail. Code using this variable might need to know it's immutable, i.e. that certain methods will just fail.

Here's the guidance from the Guava library (see: "interfaces", not implementations):
https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableCollection.html

Note that since this is just a local variable, this guidance is not super meaningful, and arguably Map<> is also acceptable in this case (that's probably how I would have written it, TBH).

Reply:
Sorry, I should reword "where possible" to "where relevant". In this particular instance, I could see an argument either way, but this is at least to point out that there are legitimate cases where you should be using the `ImmutableMap` type over `Map`.

Reply:
Got it, that makes sense. I've definitely run into cases where I had collection methods fail on me.